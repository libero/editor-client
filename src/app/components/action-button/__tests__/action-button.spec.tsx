import React from 'react';
import { create } from 'react-test-renderer';
import { ActionButton } from '../index';

describe('Action Button', () => {
  it('renders action button secondaryOutlined', () => {
    const wrapper = create(<ActionButton variant="secondaryOutlined" onClick={jest.fn()} title="secondaryOutlined" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders action button containedWarning', () => {
    const wrapper = create(<ActionButton variant="containedWarning" onClick={jest.fn()} title="containedWarning" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders action button outlinedWarning', () => {
    const wrapper = create(<ActionButton variant="outlinedWarning" onClick={jest.fn()} title="outlinedWarning" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders action button primaryContained', () => {
    const wrapper = create(<ActionButton variant="primaryContained" onClick={jest.fn()} title="primaryContained" />);
    expect(wrapper).toMatchSnapshot();
  });
});
