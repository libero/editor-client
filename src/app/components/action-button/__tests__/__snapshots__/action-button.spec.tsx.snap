// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Button renders action button containedWarning 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root makeStyles-containedRedButton-2 MuiButton-contained MuiButton-disableElevation"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    containedWarning
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Action Button renders action button outlinedWarning 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root makeStyles-outlinedRedButton-3 MuiButton-outlined MuiButton-disableElevation"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    outlinedWarning
  </span>
</button>
`;

exports[`Action Button renders action button primaryContained 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root makeStyles-regularButton-1 MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    primaryContained
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Action Button renders action button secondaryOutlined 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root makeStyles-regularButton-1 MuiButton-outlined MuiButton-disableElevation"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    secondaryOutlined
  </span>
</button>
`;
